hs_explore/management/commands/lda_recommend_resources.py

```diff
@@ -0,0 +1,177 @@
+            for candidate in user_out:
+                res_id = candidate.short_id
+                if res_id not in resource_to_keep_words:
+                    continue
+                res_keep_words = resource_to_keep_words[res_id]
+                if resource_to_published[res_id]:
+                    if len(res_keep_words) < 3:
+                        continue
+                    res_keep_words_list = list(res_keep_words)
+                    bow = dictionary.doc2bow(res_keep_words_list)
+                    t = ldamodel.get_document_topics(bow)
+                    topic_prob_dict = dict((x, y) for x, y in t)
+                    # Skip resources without any probable topics.
+                    # A probable topic is defined as any topic with probablity over (1/#topics),
+                    # which is 0.2 in our case
+                    if max(topic_prob_dict.values()) - 0.2 < 0.001:
+                        continue
+                    res_probable_topics = set()
+                    for topic, prob in topic_prob_dict.items():
+                        if prob - 0.2 > 0.001:
+                            res_probable_topics.add(topic)
+                    common_topics = set.intersection(user_probable_topics_set, res_probable_topics)
+                    # If the resource has no topics in common with the user's prbable topics set,
+                    # Skip it.
+                    if len(common_topics) == 0:
+                        continue
+
+                    for topic, words in topics_words:
+                        if topic not in common_topics:
+                            continue
+                        topic_words_set = set(words)
+                        topic_prob = 0
+                        if topic in topic_prob_dict:
+                            topic_prob = topic_prob_dict[topic]
+                        jac_sim = jaccard_sim(res_keep_words, topic_words_set)
+                        if jac_sim == 0.0:
+                            continue
+                        scaled_jac_sim = topic_prob * jac_sim
+                        if res_id not in user_to_recommend:
+                            user_to_recommend[res_id] = scaled_jac_sim
+                        else:
+                            user_to_recommend[res_id] += scaled_jac_sim
+
+            lda_users_recommendations[username] = user_to_recommend
+    user_to_recommended_resources_list = defaultdict(list)
+    for username, user_to_recommend in lda_users_recommendations.items():
+        lda_top_10_recommendations = sorted(user_to_recommend.items(), key=itemgetter(1), reverse=True)[:10]
+        for lda_res_id, lda_value in lda_top_10_recommendations:
+            if lda_res_id not in resource_to_abstract or lda_res_id not in resource_to_keep_words:
+                continue
+            user_to_recommended_resources_list[username].append((lda_res_id, lda_value))
+    print("store recommended resources")
```

Comment:
you could also put these in a logger if you'd rather.

Reply:
wait, these are management commands.  I guess a print statement is fine.

Reply:
OK